draft-ietf-privacypass-architecture.md

```diff
@@ -174,14 +132,72 @@ The following terms are used throughout this document.
+between the Client and Origin.
+- Issuance context: The interactions and set of information shared
+between the Client, Attester, and Issuer.
+- Attestation context: The interactions and set of information shared between
+the Client and Attester only, for the purposes of attesting the vailidity of the Client.
 
 # Architecture
 
 The Privacy Pass architecture consists of four logical entities --
 Client, Origin, Issuer, and Attester -- that work in concert as
-shown in {{introduction}} for token issuance and redemption. This
-section describes the purpose of token issuance and redemption
-and the requirements therein on the relevant participants.
+for token issuance and redemption. This section describes the purpose
+of token issuance and redemption and the requirements on the relevant
+participants.
+
+The typical interaction flow for Privacy Pass tokens uses the
+following steps:
+
+1. A Client interacts with an Origin by sending an HTTP request.
+The Origin sends an HTTP response that contains a token challenge
+that indicates a specific Issuer to use.
+Note that the request might be made as part of accessing a
+resource normally, or with the specific intent of triggering a token
+challenge.
+
+2. If the Client already has a token available that satisfies the token
+challenge, it can skip to step 5 and redeem its token. Otherwise, it
+invokes the issuance protocol to request a token from the designated Issuer.
+
+3. The first step in the issuance protocol is attestation. Specifically, the Attester
+performs attestation checks on the Client. These checks
+could be proof of solving a CAPTCHA, device trust, hardware attestation,
+etc (see {{attester}}).
+
+4. If attestation succeeds, the client creates a Token Request to send
+to the designated Issuer (generally via the Attester). The Attester and Issuer
+might be functions on the same server, depending on the deployment model
+(see {{deployment}}). Depending on the details of Attestation, the Client can
+send the Token Request to the Attester alongside any attestation information.
+If attestation fails, the Client receives an error and issuance aborts without a token.
+
+4. The Issuer generates a Token Response based on the Token Request, which
+is returned to the Client (generally via the Attester). Upon receiving the Token
+Response, the Client computes a token from the token challenge and Token
+Response. This token can be validated by anyone with the per-Issuer key, but
+cannot be linked to the content of the Token Request or Token Response.
+
+5. If the Client has a token, it includes it in a subsequent HTTP
+request to the Origin, as authorization. This token is sent only once.
+The Origin validates that the token was generated by the expected Issuer
+and has not already been redeemed for the corresponding token challenge.
+If the Client does not have a token, perhaps because issuance failed, the client
+does not reply to the Origin's challenge with a new request.
```

Comment:
```suggestion
2. If the Client already has a token available that satisfies the token
challenge, it can skip to step 6 and redeem its token. Otherwise, it
invokes the issuance protocol to request a token from the designated Issuer.

3. The first step in the issuance protocol is attestation. Specifically, the Attester
performs attestation checks on the Client. These checks
could be proof of solving a CAPTCHA, device trust, hardware attestation,
etc (see {{attester}}).

4. If attestation succeeds, the client creates a Token Request to send
to the designated Issuer (generally via the Attester). The Attester and Issuer
might be functions on the same server, depending on the deployment model
(see {{deployment}}). Depending on the details of Attestation, the Client can
send the Token Request to the Attester alongside any attestation information.
If attestation fails, the Client receives an error and issuance aborts without a token.

5. The Issuer generates a Token Response based on the Token Request, which
is returned to the Client (generally via the Attester). Upon receiving the Token
Response, the Client computes a token from the token challenge and Token
Response. This token can be validated by anyone with the per-Issuer key, but
cannot be linked to the content of the Token Request or Token Response.

6. If the Client has a token, it includes it in a subsequent HTTP
request to the Origin, as authorization. This token is sent only once.
The Origin validates that the token was generated by the expected Issuer
and has not already been redeemed for the corresponding token challenge.
If the Client does not have a token, perhaps because issuance failed, the client
does not reply to the Origin's challenge with a new request.
```